catalog/spatial/registry/registry-federation-admin-service-impl/src/main/java/org/codice/ddf/registry/federationadmin/service/impl/FederationAdminServiceImpl.java

```diff
@@ -66,147 +131,867 @@ public FederationAdminServiceImpl() {
         this.security = security;
     }
 
-    public void init() {
-
-    }
-
-    public void setCatalogFramework(CatalogFramework catalogFramework) {
-        this.catalogFramework = catalogFramework;
+    @Override
+    public String addRegistryEntry(String xml) throws FederationAdminException {
+        Metacard metacard = getRegistryMetacardFromString(xml);
+        return addRegistryEntry(metacard);
     }
 
-    public void setFilterBuilder(FilterBuilder filterBuilder) {
-        this.filterBuilder = filterBuilder;
+    @Override
+    public String addRegistryEntry(String xml, Set<String> destinations)
+            throws FederationAdminException {
+        Metacard metacard = getRegistryMetacardFromString(xml);
+        return addRegistryEntry(metacard, destinations);
     }
 
-    public void setParser(Parser parser) {
-        this.configurator =
-                parser.configureParser(Arrays.asList(RegistryObjectType.class.getPackage()
-                                .getName(),
-                        net.opengis.ogc.ObjectFactory.class.getPackage()
-                                .getName(),
-                        net.opengis.gml.v_3_1_1.ObjectFactory.class.getPackage()
-                                .getName()), FederationAdminServiceImpl.class.getClassLoader());
-        this.configurator.addProperty(Marshaller.JAXB_FRAGMENT, true);
-        this.parser = parser;
+    @Override
+    public String addRegistryEntry(Metacard metacard) throws FederationAdminException {
+        return addRegistryEntry(metacard, null);
     }
 
-    public void setRegistryTransformer(InputTransformer inputTransformer) {
-        this.registryTransformer = inputTransformer;
+    @Override
+    public String addRegistryEntry(Metacard metacard, Set<String> destinations)
+            throws FederationAdminException {
+        validateRegistryMetacard(metacard);
+
+        String metacardId;
+
+        Subject systemSubject = security.getSystemSubject();
+        Map<String, Serializable> properties = new HashMap<>();
+        properties.put(SecurityConstants.SECURITY_SUBJECT, systemSubject);
+        CreateRequest createRequest = new CreateRequestImpl(Collections.singletonList(metacard),
+                properties,
+                destinations);
+
+        try {
+            CreateResponse response = catalogFramework.create(createRequest);
+            metacardId = response.getCreatedMetacards()
+                    .get(0)
+                    .getId();
+        } catch (IngestException | SourceUnavailableException e) {
+            throw new FederationAdminException("Error adding local registry entry.", e);
```

Comment:
I can't remember if I already asked this before, but is OK to do? 
Shouldn't we just take the exception we get instead of throwing another?